ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/VertxNonceRequestTransmitter.java

```diff
@@ -115,10 +116,17 @@ private void handleResponse(final Buffer bodyBuffer, final CompletableFuture<Big
       }
       result.completeExceptionally(new RuntimeException("Web3 did not provide a string response."));
     } catch (final DecodeException e) {
-      result.completeExceptionally(
-          new RuntimeException(
-              "Web3 Provider did not respond with a valid success message: "
-                  + bodyBuffer.toString(UTF_8)));
+      result.completeExceptionally(new JsonRpcException(determineErrorCode(bodyBuffer)));
```

Comment:
Is it ever possible to determine the error from the response if a decode exception has occurred? Looks like determineErrorCode will always return a JsonRpcError.INTERNAL_ERROR when called from this catch?

Reply:
This function attempts to deserialise the body as a JsonRpcSUCCESSResponse - determineErrorCode attempts to decode as an JsonRpcERRORResponse - which should contain a error code which can then be used to populate this field.
Does it work as expected? I _thought_ this was proven in the Private Txn nonce issue ... :/